src/gsm.c

```diff
@@ -152,18 +152,17 @@ int32_t gsm_select_res(swicc_fs_st const *const fs,
             10U; /* Everything except the last portion which is optional. */
 
         /* GSM specific data. */
-        /* "File characteristics." */
+        /* "File characteristics." (TS 51.011 Rel4 9.2.1 Detail 1) */
         /**
          * LSB>MSB
          *    1b Clock stop             = 0 (not allowed)
          *  + 1b Auth. alg. clock freq. = 1 (13/4 MHz)
          *  + 1b Clock stop             = 0 (high not preferred)
          *  + 1b Clock stop             = 0 (low not preferred)
-         *  + 1b 0 (from GSM 11.12 @todo Look into GSM 11.12)
-         *  + 2b RFU                    = 00
+         *  + 3b Supply voltage ind.    = 3 (1.8V Technology SIM (supports 3V/5V)
```

Comment:
I set this to `0b000` because per ETSI TS 102 221 (v16.4.0) 6.2.1, if the card supports multiple supply voltage classes, they need to be consecutive, i.e., AB, BC, ABC, but not AC. Setting this to `0b011` would mean classes A and B would appear unsupported? Is there a mention in any of the specs about support for classes A and B when only C is indicated? Technically, ETSI TS 102 221 also extends the ISO classes with D and E. In summary, I don't know what indication to use, in order to support all the supply voltage classes and maximize compatibility.

Missing ')', and I'd prefer if you swapped `3` to binary `011` as the other flags are all listed in binary.

Reply:
I've fixed the parenthesis and the decimal/binary in my updated branch/PR.

Regarding your reference to ETSI TS 102 221 - I don't really understand why it would apply here. Classic GSM SIM is not an UICC application. So unless TS 51.011 explicitly (directly or indirectly e.g. via 31.101) references ETSI TS 102 221, I don't see why or how it would apply.  Particularly  since TS 51.011 gives its very own  definition of those bits. Also, the value `011` was taken from a real-world T-Mobile SIM card, which works in all the phones I've tested.

Reply:
You're absolutely right. I also checked in GSM 11.12, and it seems 3v SIM cards must also support 5v, which should increase compatibility.